widget/diagramwidget/diagram.go

```diff
@@ -47,6 +49,14 @@ type DiagramWidget struct {
 	diagramElementLinkDependencies map[string][]linkPadPair
 	// ConnectionTransaction holds transient data during the creation of a link. It is public for testing purposes
 	ConnectionTransaction *ConnectionTransaction
+	// AnchoredTextChangedCallback is called when the position, offset, or text of the anchored text has changed
+	AnchoredTextChangedCallback func(*AnchoredText)
+	// DragEndCallback is called when a dragEnd event occurs in the drawing area
+	DragEndCallback func()
```

Comment:
These are pretty much duplicating the existing Fyne events for `DragEnd` etc - can they be exposed in the way of other widgets such that someone extending the widget can do that and replace the method, delegating back upward to the embedded widget?